var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -717,6 +743,20 @@ def define_tpl(trilinos_name, spack_name, have_dep):
                     for arch in spec.variants['cuda_arch'].value
                 )
 
+            amd_microarches = []
+            if "+rocm" in spec:
+                options.append(define('Kokkos_ENABLE_ROCM', False))
+                options.append(define('Kokkos_ENABLE_HIP', True))
+                if '+tpetra' in spec:
+                    options.append(define('Tpetra_INST_HIP', True))
+                for amdgpu_target in spec.variants['amdgpu_target'].value:
+                    if amdgpu_target != "none":
+                        if amdgpu_target in self.amdgpu_arch_map:
+                            amd_microarches.append(
+                                self.amdgpu_arch_map[amdgpu_target])
+                for arch in amd_microarches:
+                    options.append(self.define("Kokkos_ARCH_" + arch.upper(), True))
```

Comment:
Maybe somewhere between there and here (since 'none' isn't in `amdgpu_arch_map` and all other architectures are excluded with `conflicts`, the two `if` statements are redundant):
```suggestion
              if "+rocm" in spec:
                define_kok_enable('ROCM', False)
                define_kok_enable('HIP', True)
                if '+tpetra' in spec:
                    options.append(define('Tpetra_INST_HIP', True))
                for tgt in spec.variants['amdgpu_target'].value:
                    try:
                        arch = self.amdgpu_arch_map[amdgpu_target]
                    except KeyError:
                        pass
                    else:
                        options.append(define("Kokkos_ARCH_" + arch.upper(), True))
```

Reply:
Do we still have?

        tpl_variant_map = [
            ('ADIOS2', 'adios2'),
            ('Boost', 'boost'),
            ('CUDA', 'cuda'),
            **('ROCm', 'rocm'),**
            ('HDF5', 'hdf5'),
            ('HYPRE', 'hypre'),
            ('MUMPS', 'mumps'),
            ('UMFPACK', 'suite-sparse'),
            ('SuperLU', 'superlu'),
            ('SuperLUDist', 'superlu-dist'),
            ('X11', 'x11'),
        ]

Reply:
It looks like rocm is a Kokkos option, not a trilinos option 😒

Reply:
Yes. I noticed. Kokkos and Trilinos teams do not agree how to call AMD GPU backend...  This is confusing.